test/integration/tcx_test.go

```diff
@@ -0,0 +1,59 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	tcxGoCounterKustomize       = "https://github.com/bpfman/bpfman/examples/config/default/go-tcx-counter/?timeout=120&ref=main"
+	tcxGoCounterUserspaceNs     = "go-tcx-counter"
+	tcxGoCounterUserspaceDsName = "go-tcx-counter-ds"
+)
+
+func TestTcxGoCounter(t *testing.T) {
+	t.Log("deploying tcx counter program")
+	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), tcxGoCounterKustomize))
+	addCleanup(func(context.Context) error {
+		cleanupLog("cleaning up tcx counter program")
+		return clusters.KustomizeDeleteForCluster(ctx, env.Cluster(), tcxGoCounterKustomize)
+	})
+
+	t.Log("waiting for go tcx counter userspace daemon to be available")
+	require.Eventually(t, func() bool {
+		daemon, err := env.Cluster().Client().AppsV1().DaemonSets(tcxGoCounterUserspaceNs).Get(ctx, tcxGoCounterUserspaceDsName, metav1.GetOptions{})
+		require.NoError(t, err)
+		return daemon.Status.DesiredNumberScheduled == daemon.Status.NumberAvailable
+	},
+		// Wait 5 minutes since cosign is slow, https://github.com/bpfman/bpfman/issues/1043
+		5*time.Minute, 10*time.Second)
+
+	pods, err := env.Cluster().Client().CoreV1().Pods(tcxGoCounterUserspaceNs).List(ctx, metav1.ListOptions{LabelSelector: "name=go-tcx-counter"})
+	require.NoError(t, err)
+	gotcCounterPod := pods.Items[0]
+
+	req := env.Cluster().Client().CoreV1().Pods(tcxGoCounterUserspaceNs).GetLogs(gotcCounterPod.Name, &corev1.PodLogOptions{})
+
+	require.Eventually(t, func() bool {
+		logs, err := req.Stream(ctx)
+		require.NoError(t, err)
+		defer logs.Close()
+		output := new(bytes.Buffer)
+		_, err = io.Copy(output, logs)
+		require.NoError(t, err)
+		t.Logf("counter pod log %s", output.String())
+
+		return doTcCheck(t, output)
```

Comment:
doTcxCheck?

Reply:
It seems like we need a doTcxCheck() function.  This is probably working because the tc test is also running and there are logs from that test.

Reply:
I reused the Tc check since both look for the same string might be better to have one for tcx will do

Reply:
Actually, I just looked, and it doesn't.  I think it should, though.  That's probably another reason I changed the TC, TCX, and XDP logs in the app counter example.

Reply:
We can do that in a separate pr, though.

Reply:
Or not.  They're all in separate pods, so I guess it's clear.  Sorry for the stream of consciousness as I remember how all this stuff works :-).